### DISCUSSION TITLE
useFieldArray not adding items

### DISCUSSION THREAD

**Discussion by PointSingularity** [react-hook-form/react-hook-form#9227] [Q&A] [answered]
Here is the [code](https://codesandbox.io/s/mystifying-thunder-y9hl5d?file=/src/BaseForm.tsx).

Basically, I have a custom number input control. I want to use the useFieldArray to add new input fields.

The problem is if the fields array is empty, when I click append, nothing happens.
If the first item is defined, then it works, but only until there are no items left in the array.

You can uncomment the values for the array to see it working.

I am not sure what I am doing wrong, any insight would be helpful.

Thank you in advance!

**Comment by Moshyfawn** [comment, marked as answer]
You can't init the same field array in 2 different places. You're creating 2 separate arrays doing this.

There's an RFC #7544 to include a field array context that would allow you to use `useFieldArray` in a similar manner you're currently trying to use it.

**Reply by Moshyfawn** [reply]
In your use-case, I don't see why you wouldn't simply pass the first field empty field array as default value to avoid creating the same field array in 2 different places.

**Reply by PointSingularity** [reply]
This was indeed the problem, I didn't know this caveat. I'll try to read more next time 😅

I now just pass the form and array as props (code sandbox is also updated):

```tsx
<FirstGeometryPartForm form={form} array={geometryParts} />
<GeometryPartsForm form={form} array={geometryParts} />
```



Thank you!
